esphome/components/online_image/online_image.h

```diff
@@ -122,6 +124,7 @@ class OnlineImage : public PollingComponent,
   image::Image *placeholder_{nullptr};
 
   std::string url_{""};
+  std::map<const char *, std::function<optional<const char *>()>> headers_{};
```

Comment:
```suggestion
  std::map<const char *, TemplatableValue<std::string>> headers_{};
```
`TemplatableValue` is a utility class in ESPHome for just this application.